app/validators/licence_identifier_unique_validator.rb

```diff
@@ -0,0 +1,15 @@
+class LicenceIdentifierUniqueValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if value.blank?
+
+    existing_licence_identifiers = []
+
+    record.class.find_each do |licence|
```

Comment:
How fast is this? Is it making 309 api calls every time we try to save a licence?

Reply:
(Update - it's more like 8 API calls, as they're done in batches of 50, and there isn't currently a good optimisation we can do here. Appears to be quick in integration, we should just accept this is not perfect but necessary and keep an eye on whether it causes slowdowns in use. Marking as resolved).